pkg/controller/atlasproject/project.go

```diff
@@ -31,7 +33,9 @@ func ensureProjectExists(connection atlas.Connection, project *mdbv1.AtlasProjec
 				}
 
 			}*/
-		return err
+		return "", workflow.Terminate(workflow.ProjectNotCreatedInAtlas, err.Error())
 	}
-	return nil
+	ctx.EnsureStatusOption(status.NewIDOption(p.ID))
```

Comment:
this is an example of how the "internal" method can update the "ongoing" status. Here the method is responsible for ensuring the project ID is added to the status in the end